### PR TITLE
OTel-Arrow exporter timeout propagation

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** Exporter side of https://github.com/open-telemetry/otel-arrow/issues/227. The receiver side is https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34742.

**Link to tracking Issue:** https://github.com/open-telemetry/otel-arrow/issues/227

**Testing:** Adds a test for the expected metadata propagation.

**Documentation:** Since this is expected of gRPC receivers, no docs are changed.

### CODE REVIEW

File: exporter/otelarrowexporter/internal/arrow/exporter_test.go [hunk 1]
```diff
@@ -576,65 +577,92 @@ func TestArrowExporterStreaming(t *testing.T) {
...
+					dead := out.Get("grpc-timeout")
+					require.Equal(t, 1, len(dead))
+					require.Less(t, 0, len(dead[0]))
+					to, err := grpcutil.DecodeTimeout(dead[0])
+					require.NoError(t, err)
+					// Allow the test to lapse for 0.5s.
+					require.Less(t, time.Second/2, to)
+					require.GreaterOrEqual(t, time.Second, to)
+					out.Delete("grpc-timeout")
+				}
```
Comment (jmacd) [inline exporter/otelarrowexporter/internal/arrow/exporter_test.go:659]: For the most part, I just re-indented this test. This is all that really changes.

**Review by mwear — APPROVED** [review]

**Review by moh-osman3 — APPROVED** [review]
LGTM

**Comment by jmacd** [comment]
This has been approved for 2 weeks. Can it merge?

**Comment by MovieStoreGuy** [comment]
> This has been approved for 2 weeks. Can it merge?

Can I ask you to fix up the merge conflicts please?

**Comment by jmacd** [comment]
> Can I ask you to fix up the merge conflicts please?

Of course! I fix the conflicts after every release, but this has been sitting for two release cycles and now I'll have to fix them twice.

**Review by songy23 — COMMENTED** [review]
> Replace statements are out of date, please run "make crosslink" and commit the changes in this PR.

Please fix linter as well https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10704184781/job/29676638560?pr=34733

**Comment by jmacd** [comment]
@songy23 Thank you. I appreciate your help. The linter changed while this PR was open,.

I want to point out that this PR was approved and green following commit [f735180](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34733/commits/f735180df61fd28336a7aceaec323e6a1e68515f) and again following commit [9e4d8e3](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34733/commits/9e4d8e35bae1069c7f725db3dae29ab23e2adb3b). It takes a relatively large amount of work to keep these PRs green, and this is the third time. Please merge!

**Review by songy23 — APPROVED** [review]
cc @open-telemetry/collector-contrib-maintainers  this one should be good to go

**Review by codeboten — APPROVED** [review]

**Pull request merged**